src/Untagged/TypeCheck.purs

```diff
@@ -58,7 +58,7 @@ instance hasRuntimeTypeRecord ::
   ) => HasRuntimeType {|r} where
   hasRuntimeType _ x = hasJsType "object" x && hasRuntimeTypeRecRL (RLProxy :: _ rl) (unsafeToForeign x)
 
-class HasRuntimeTypeRecordRL (rl :: RowList) where
+class HasRuntimeTypeRecordRL (rl :: RowList Type) where
   hasRuntimeTypeRecRL :: RLProxy rl -> Foreign -> Boolean
```

Comment:
It's not necessary, but the policy we've taken in core libraries is to:

1. Switch all proxy types in function signatures / class members from `SProxy x` or `RLProxy x` or `RProxy x` etc. to all be `forall proxy. proxy x`, which allows it to be used with any proxy type
2. Switch all concrete uses of a proxy to be `Proxy`. Ie. `x = SProxy :: SProxy "foo"` becomes `x = Proxy :: Proxy "foo"`. Instance heads also are switched to use `Proxy`.

This isn't strictly necessary because these proxies are deprecated, not removed. But they'll be removed in PureScript 0.15, so this gets ahead of some upcoming breaking changes.

Reply:
You could see https://github.com/garyb/purescript-codec-argonaut/pull/35 as an example if you're interested!